fetchmedia.go

```diff
@@ -74,22 +74,25 @@ func (g grabber) fetchMedia(mediaSource, mediaFolderPath, mediaFilename string)
 
 	}
 	defer source.Close()
-	// Defining two buffers
-	var buffer bytes.Buffer
 
-	// Calling MultiWriter method with its parameters
-	writer := io.MultiWriter(w, &buffer)
-	_, err = io.Copy(writer, source)
+	_, err = io.Copy(w, source)
 	if err != nil {
 		// There shouldn't be any problem with the writer, but the reader
 		// might have an issue
 		return "", &FileRetrievalError{Source: mediaSource, Err: err}
 	}
+
 	// Detect the mediaType
-	mime, err := mimetype.DetectReader(&buffer)
+	r, err := filesystem.Open(mediaFilePath)
+	if err != nil {
+		return "", err
```

Comment:
I'm not sure if this should be `return "", err`, `return "", &FileRetrievalError{Source: mediaSource, Err: err}`, or `panic(err)` - please advise

Reply:
I guess that panic should be raised.
But I wonder why you change this method?
What’s wrong with the multiwriter?

Reply:
The multi-writer wrote to `bytes.Buffer` which is held in-memory and can cause out-of-memory crashes with large files such as videos. This switches to a temporary file which should be capable of handling large files.